sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -343,31 +343,55 @@ export class EventHubSender extends LinkEntity {
   }
   /**
    * Returns maximum message size on the AMQP sender link.
-   * @ignore
+   * @param abortSignal An implementation of the `AbortSignalLike` interface to signal the request to cancel the operation.
+   * For example, use the &commat;azure/abort-controller to create an `AbortSignal`.
    * @returns Promise<number>
+   * @throws {AbortError} Thrown if the operation is cancelled via the abortSignal.
    */
-  async getMaxMessageSize(): Promise<number> {
-    try {
-      if (!this.isOpen()) {
-        log.sender(
-          "Acquiring lock %s for initializing the session, sender and " +
-            "possibly the connection.",
-          this.senderLock
+  async getMaxMessageSize(abortSignal?: AbortSignalLike): Promise<number> {
+    return new Promise<number>(async (resolve, reject) => {
```

Comment:
The cancellable part of `getMaxMessageSize` is the link creation part. Everything else is syncronous.
So, I would suggest doing an early exit before creating the above promise

```javascript
if (this.isOpen()) {
    return this._sender!.maxMessageSize;
}
```